app/services/stage_files.rb

```diff
@@ -6,17 +6,34 @@ def self.stage(signed_ids, druid)
     # Skip side effects if no signed IDs provided
     return yield if signed_ids.empty?
 
+    @dir = StagingDirectory.new(druid: druid, staging_location: Settings.staging_location)
+
     blobs = Blobs.blobs_for(signed_ids)
```

Comment:
This is the line that is causing the `Blobs` class to have to be aware of whether it's getting legit signed IDs or not, right? I wonder if `StageFiles` could have the added smarts instead, so that we can better contain the concern about "am I getting signed IDs or other IDs?" rather than having it leak out.

Reply:
This was less about legit IDs and the order things happened in. It currently lives in `copy_files_to_staging` so the method it was in was mocked, and thus this was skipped, we can't do that with the need for blob vs. globus as far as I can tell.